app/src/main/java/com/example/android/codelabs/paging/data/GithubPagingSource.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.android.codelabs.paging.data
+
+import androidx.paging.PagingSource
+import com.example.android.codelabs.paging.api.GithubService
+import com.example.android.codelabs.paging.api.searchRepos
+import com.example.android.codelabs.paging.model.Repo
+import com.example.android.codelabs.paging.model.RepoSearchResult
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+
+@ExperimentalCoroutinesApi
+@FlowPreview
+class GithubPagingSource(
+        private val service: GithubService,
+        private val query: String
+) : PagingSource<Int, Repo>() {
+    override suspend fun load(params: LoadParams<Int>): LoadResult<Int, Repo> {
+        val position = params.key ?: 0
+        val apiResponse = searchRepos(service, query, position, GithubRepository.NETWORK_PAGE_SIZE)
```

Comment:
should sync with Dustin in the PagingWithNetworkSample to standardize as to whether we want to use suspend Response<Foo> retrofit APIs, or the Response-less ones. I'm fine either way, but there is some value in consistency.